src/Client.php

```diff
@@ -138,17 +214,77 @@ public function createRefund(Refund $refund)
      * @param string $id Refund id
      * @return Refund
      * @throws \GuzzleHttp\Exception\GuzzleException
-     * @throws ValidationException
+     * @throws ValidationException|CertificateException
      */
     public function getRefund($id)
     {
-        $response = $this->sendRequest('GET', '/refunds/' . $id);
+        $response = $this->sendRequest('GET', 'v1/refunds/' . $id);
 
         return new Refund(
-            json_decode((string) $response->getBody(), true)
+            json_decode((string)$response->getBody(), true)
+        );
+    }
+
+    /**
+     * Create a Swish payout request.
+     *
+     * @param PayoutRequest $payoutRequest
+     * @return string
+     * @throws CertificateException
+     * @throws GuzzleException
+     * @throws ValidationException
+     */
+    public function createPayoutRequest(PayoutRequest $payoutRequest)
+    {
+        $response = $this->sendRequest('POST', 'v1/payouts', [
+            'json' => $this->filterRequestBody((array)$payoutRequest),
+        ]);
+
+        return Util::getObjectIdFromResponse($response);
+    }
+
+    /**
+     * Get the current status from a given payout request.
+     *
+     * @param $payoutInstructionUUID - The initially set Payout Instruction UUID to query.
+     * @return PayoutRequest
+     * @throws CertificateException
+     * @throws GuzzleException
+     * @throws ValidationException
+     */
+    public function getPayoutRequest($payoutInstructionUUID)
+    {
+        $response = $this->sendRequest('GET', 'v1/payouts/' . $payoutInstructionUUID);
+
+        return new PayoutRequest(
+            json_decode((string)$response->getBody(), true)
         );
     }
 
+    /**
+     * Generate a QR Code for a given payment request token.
+     * @param QRCodeRequest $QRCodeRequest
+     * @return \Psr\Http\Message\StreamInterface
+     * @throws CertificateException
+     * @throws GuzzleException
+     * @throws ValidationException
+     */
+    public function createQRCode(QRCodeRequest $QRCodeRequest)
+    {
+        $response = $this->sendRequest('POST', 'v1/commerce', [
+            'json' => $this->filterRequestBody((array)$QRCodeRequest),
+        ]);
+        return $response->getBody();
+    }
+
+    /**
+     * @return string - A time based UUID
+     */
+    private function generateUUID()
+    {
+        return strtoupper(md5(time()));
```

Comment:
Hehe nice try but this doesn't look like an UUID at all. Is this some kind of hard swish requirement? If not, we can probably generate a random V4 UUID.